### PR TITLE
nixos/xterm: Set default to xserver.enable

### LINKED ISSUE #69079: Unstable channel hasn't updated for 10+ days
It's been a while since we've had an update of the `nixos-unstable` channel. According to https://hydra.nixos.org/job/nixos/trunk-combined/tested#tabs-constituents, there have been plenty of problems with Hydra, but the failure of [nixos.tests.containers-imperative.x86_64-linux](https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.containers-imperative.x86_64-linux) seems legit. Does anyone know why that test is broken?

### PR DESCRIPTION
**Pull request by alexarice**
###### Motivation for this change

Fixes #69079
I bisected this issue down to https://github.com/NixOS/nixpkgs/commit/456c42c3e8787d83d577526af90263de9b3d512d
This changes the default for xterm to be true if stateVersion is older than 19.09.
I believe that in this case the default should be set to the xserver.enable default, as it was 2 commits before.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @worldofpeace

### CODE REVIEW

**Comment by matthewbauer** [comment]
We may want to just enable xterm on those tests that need it.

**Comment by alexarice** [comment]
The problem is that it is enabled when it shouldn't be (i.e. when xserver is not enabled). The tests don't need it

**Comment by matthewbauer** [comment]
> The problem is that it is enabled when it shouldn't be (i.e. when xserver is not enabled). The tests don't need it

Ah ok! My bad, I misunderstood what the problem.

**Comment by worldofpeace** [comment]
Yeah this is what I meant to fix after it got merged a bit back, but I forgot about it :rofl: 

I actually want a fix like https://github.com/NixOS/nixpkgs/pull/68371, but I'm not too sure about the PR because I don't understand what we need to support.

**Comment by FRidh** [comment]
Let's get this in first to unblock the channels.

**Review by worldofpeace — APPROVED** [review]
LGTM

**Pull request merged**
